packages/server-wallet/src/wallet/multi-threaded-wallet/manager.ts

```diff
@@ -67,17 +78,23 @@ export class WorkerManager {
 
     const worker = await this.pool.acquire().promise;
 
-    const resultPromise = new Promise<T>((resolve, reject) =>
-      worker.once('message', (response: Either<Error, T>) => {
+    const resultPromise = new Promise<T>((resolve, reject) => {
+      const handler = (response: Either<Error, T>) => {
         this.pool?.release(worker);
 
         if (isLeft(response)) {
           reject(response.left);
-        } else {
+        } else if (isRight(response)) {
+          // WARNING: The correctness of this implementation relies on the fact that a worker
```

Comment:
It might be worth putting a similar warning in `worker.ts` where we construct the `SingleThreadedWallet`. That way if someone tries changing it to a `MultiThreadedWallet` they see the warning as well.

Reply:
I don't actually think there's a worry in `worker.ts`. A worker receives an operation and sends back the result. It's up to the manager to determine that the result message corresponded to the API call.

> That way if someone tries changing it to a MultiThreadedWallet they see the warning as well.

I don't quite understand: you mean if someone changed `SingleThreadedWallet` to `MultiThreadedWallet` in `worker.ts`? That would be a pretty flawed decision that the `worker_threads` API might not even allow -- can worker thread spawn worker threads?. I don't know that we want to protect against that with a comment?

Reply:
My thinking was that it might be easy to miss the warning in `sendOperation`?  But that might be overkill.